pkg/apis/core/validation/shoot.go

```diff
@@ -1148,20 +1145,59 @@ func ValidateClusterAutoscaler(autoScaler core.ClusterAutoscaler, fldPath *field
+		cloudProfileReference = &core.CloudProfileReference{
+			Kind: "CloudProfile",
+			Name: *cloudProfileName,
+		}
+	}
+	if oldCloudProfileReference == nil && oldCloudProfileName != nil {
+		oldCloudProfileReference = &core.CloudProfileReference{
+			Kind: "CloudProfile",
+			Name: *oldCloudProfileName,
+		}
+	}
+
+	// Ensure that cloudProfileReference is provided and contains a cloudprofile reference of a valid kind, else fail early.
+	// Due to the field synchronization in the shoot strategy, it is safe to only check the cloudProfileReference.
+	if cloudProfileReference == nil || len(cloudProfileReference.Name) == 0 {
+		return append(allErrs, field.Required(fldPath.Child("name"), "must specify a cloud profile"))
 	}
-	cloudProfileReferenceKind := sets.New("CloudProfile", "NamespacedCloudProfile")
+	cloudProfileReferenceKind := sets.New(v1beta1constants.CloudProfileReferenceKindCloudProfile, v1beta1constants.CloudProfileReferenceKindNamespacedCloudProfile)
 	if !cloudProfileReferenceKind.Has(cloudProfileReference.Kind) {
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("kind"), cloudProfileReference.Kind, sets.List(cloudProfileReferenceKind)))
 	}
-	if len(cloudProfileReference.Name) == 0 {
-		allErrs = append(allErrs, field.Required(fldPath.Child("name"), "must specify a CloudProfile or NamespacedCloudProfile name"))
+
+	// Fail early for changing a NamespacedCloudProfile to a CloudProfile
+	if oldCloudProfileReference != nil && oldCloudProfileReference.Kind == v1beta1constants.CloudProfileReferenceKindNamespacedCloudProfile &&
+		(cloudProfileReference.Kind != v1beta1constants.CloudProfileReferenceKindNamespacedCloudProfile) {
+		return append(allErrs, field.Forbidden(fldPath.Child("kind"), "a namespacedcloudprofile must not be changed back to a cloudprofile"))
 	}
 
+	// Grant directly for changing a CloudProfile to a NamespacedCloudProfile as the admission validation ensures
+	// that the NamespacedCloudProfile is a direct descendant to the CloudProfile.
+	if (oldCloudProfileReference == nil || oldCloudProfileReference.Kind == v1beta1constants.CloudProfileReferenceKindCloudProfile) &&
+		(cloudProfileReference.Kind == v1beta1constants.CloudProfileReferenceKindNamespacedCloudProfile) {
+		return allErrs
+	}
+
+	// Else, ensure that the name of the (Namespaced)CloudProfile stays the same
+	// Due to the field synchronization in the shoot strategy it is safe to only check in cloudProfileReference
+	var formerCloudProfileName *string = oldCloudProfileName
+	if oldCloudProfileReference != nil {
+		formerCloudProfileName = &oldCloudProfileReference.Name
+	}
+	if formerCloudProfileName == nil {
+		return allErrs
+	}
+	if *formerCloudProfileName != cloudProfileReference.Name {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("name"), "changing the cloudProfile name is not allowed, except for switching from a CloudProfile to a directly descendant NamespacedCloudProfile"))
+	}
```

Comment:
```suggestion
	if formerCloudProfileName != nil && *formerCloudProfileName != cloudProfileReference.Name {
		allErrs = append(allErrs, field.Forbidden(fldPath.Child("name"), "changing the cloudProfile name is not allowed, except for switching from a CloudProfile to a directly descendant NamespacedCloudProfile"))
	}
```

Nit, to be more concise.